MSSQL bug involving `MinimumOf` and `CombineAsSet`

Originally found by the gentests in this failed run:
https://github.com/opensafely-core/ehrql/actions/runs/7255717313/job/19766828683

<details>
<summary>Failing example (obviously not yet shrunk):</summary>

```py
import datetime

from ehrql.query_model.nodes import (
    AggregateByPatient,
    Case,
    Column,
    Filter,
    Function,
    InlinePatientTable,
    PickOneRowPerPatient,
    Position,
    SelectColumn,
    SelectTable,
    Sort,
    TableSchema,
    Value,
)
from tests.generative.test_query_model import data_setup


e0 = SelectTable(
    "e0",
    TableSchema(
        i1=Column(int, constraints=()),
        i2=Column(int, constraints=()),
        b1=Column(bool, constraints=()),
        b2=Column(bool, constraints=()),
        d1=Column(datetime.date, constraints=()),
        d2=Column(datetime.date, constraints=()),
        f1=Column(float, constraints=()),
        f2=Column(float, constraints=()),
        s1=Column(str, constraints=()),
        s2=Column(str, constraints=()),
    ),
)

e0_b2 = SelectColumn(e0, "b2")

e0_i2 = SelectColumn(e0, "i2")

inline_0 = InlinePatientTable(
    (
        (
            9,
            1,
            6,
            False,
            False,
            datetime.date(2019, 4, 15),
            datetime.date(2017, 1, 10),
            1.099609375,
            0.333251953125,
            "c",
            "bcc",
        ),
        (
            2,
            7,
            4,
            True,
            False,
            datetime.date(2012, 1, 6),
            datetime.date(2014, 7, 1),
            1.900390625,
            1.5,
            "",
            "c",
        ),
        (
            7,
            8,
            5,
            True,
            False,
            datetime.date(2013, 1, 4),
            datetime.date(2018, 12, 4),
            9.53125,
            0.0,
            "aac",
            "cb",
        ),
        (
            6,
            6,
            9,
            True,
            True,
            datetime.date(2017, 9, 1),
            datetime.date(2020, 6, 16),
            0.155029296875,
            0.5,
            "a",
            "abc",
        ),
        (
            1,
            4,
            7,
            True,
            True,
            datetime.date(2018, 8, 11),
            datetime.date(2018, 12, 15),
            0.0,
            0.5,
            "",
            "bca",
        ),
        (
            4,
            4,
            0,
            False,
            False,
            datetime.date(2016, 5, 10),
            datetime.date(2013, 12, 2),
            11.0,
            0.0,
            "ccc",
            "",
        ),
        (
            5,
            0,
            10,
            True,
            False,
            datetime.date(2020, 12, 16),
            datetime.date(2011, 3, 24),
            5.960464477539063e-08,
            5.960464477539063e-08,
            "cba",
            "bcc",
        ),
        (
            8,
            3,
            9,
            False,
            False,
            datetime.date(2016, 8, 20),
            datetime.date(2018, 7, 5),
            1.1920928955078125e-07,
            5.2109375,
            "",
            "acc",
        ),
        (
            10,
            2,
            9,
            False,
            False,
            datetime.date(2016, 2, 4),
            datetime.date(2013, 2, 9),
            0.0,
            10.4375,
            "bcc",
            "a",
        ),
        (
            3,
            3,
            10,
            False,
            True,
            datetime.date(2013, 12, 28),
            datetime.date(2019, 5, 31),
            0.0,
            0.0,
            "bb",
            "",
        ),
    ),
    TableSchema(
        i1=Column(int, constraints=()),
        i2=Column(int, constraints=()),
        b1=Column(bool, constraints=()),
        b2=Column(bool, constraints=()),
        d1=Column(datetime.date, constraints=()),
        d2=Column(datetime.date, constraints=()),
        f1=Column(float, constraints=()),
        f2=Column(float, constraints=()),
        s1=Column(str, constraints=()),
        s2=Column(str, constraints=()),
    ),
)

inline_0_d2 = SelectColumn(inline_0, "d2")

e1 = SelectTable(
    "e1",
    TableSchema(
        i1=Column(int, constraints=()),
        i2=Column(int, constraints=()),
        b1=Column(bool, constraints=()),
        b2=Column(bool, constraints=()),
        d1=Column(datetime.date, constraints=()),
        d2=Column(datetime.date, constraints=()),
        f1=Column(float, constraints=()),
        f2=Column(float, constraints=()),
        s1=Column(str, constraints=()),
        s2=Column(str, constraints=()),
    ),
)

e1_b2 = SelectColumn(e1, "b2")

inline_0_d1 = SelectColumn(inline_0, "d1")

inline_0_i2 = SelectColumn(inline_0, "i2")

e1_s1 = SelectColumn(e1, "s1")

e1_s2 = SelectColumn(e1, "s2")

inline_1 = InlinePatientTable(
    (
        (
            9,
            8,
            2,
            True,
            False,
            datetime.date(2020, 8, 31),
            datetime.date(2013, 12, 2),
            0.333251953125,
            0.0,
            "cca",
            "aca",
        ),
        (
            5,
            5,
            6,
            False,
            True,
            datetime.date(2018, 2, 11),
            datetime.date(2017, 12, 12),
            1.099609375,
            0.5,
            "aa",
            "b",
        ),
        (
            3,
            4,
            5,
            False,
            True,
            datetime.date(2013, 9, 8),
            datetime.date(2019, 6, 28),
            1.84375,
            0.0,
            "",
            "c",
        ),
        (
            1,
            6,
            2,
            False,
            True,
            datetime.date(2011, 1, 7),
            datetime.date(2015, 2, 7),
            0.0,
            6.103515625e-05,
            "ca",
            "c",
        ),
        (
            7,
            8,
            5,
            True,
            False,
            datetime.date(2012, 1, 29),
            datetime.date(2013, 12, 23),
            0.0,
            1.5,
            "ca",
            "baa",
        ),
        (
            2,
            4,
            1,
            True,
            False,
            datetime.date(2013, 1, 15),
            datetime.date(2019, 2, 16),
            6.103515625e-05,
            1.1920928955078125e-07,
            "b",
            "",
        ),
    ),
    TableSchema(
        i1=Column(int, constraints=()),
        i2=Column(int, constraints=()),
        b1=Column(bool, constraints=()),
        b2=Column(bool, constraints=()),
        d1=Column(datetime.date, constraints=()),
        d2=Column(datetime.date, constraints=()),
        f1=Column(float, constraints=()),
        f2=Column(float, constraints=()),
        s1=Column(str, constraints=()),
        s2=Column(str, constraints=()),
    ),
)

inline_1_i1 = SelectColumn(inline_1, "i1")

population = Function.GE(
    Value(8),
    Function.MonthFromDate(
        AggregateByPatient.Max(
            SelectColumn(
                Filter(
                    Filter(Filter(e0, e0_b2), Function.GT(Value(1), e0_i2)),
                    Function.LE(
                        Value(6.103515625e-05),
                        SelectColumn(
                            Filter(Filter(e0, e0_b2), Function.GT(Value(1), e0_i2)),
                            "f2",
                        ),
                    ),
                ),
                "d1",
            )
        )
    ),
)

variable = Function.MinimumOf(
    (
        Function.ToFirstOfYear(
            Function.ToFirstOfMonth(
                Function.MaximumOf(
                    (
                        Function.DateAddYears(
                            Function.ToFirstOfMonth(inline_0_d2), Value(8)
                        ),
                        Function.DateAddYears(
                            Value(datetime.date(2020, 11, 28)),
                            Function.DateDifferenceInMonths(
                                SelectColumn(
                                    PickOneRowPerPatient(
                                        Filter(
                                            Filter(
                                                Sort(
                                                    Filter(
                                                        Filter(
                                                            Filter(
                                                                Filter(e1, e1_b2), e1_b2
                                                            ),
                                                            SelectColumn(
                                                                Filter(e1, e1_b2), "b1"
                                                            ),
                                                        ),
                                                        SelectColumn(
                                                            Filter(e1, e1_b2), "b2"
                                                        ),
                                                    ),
                                                    SelectColumn(
                                                        Filter(
                                                            Filter(
                                                                Filter(e1, e1_b2), e1_b2
                                                            ),
                                                            SelectColumn(
                                                                Filter(e1, e1_b2), "b1"
                                                            ),
                                                        ),
                                                        "s1",
                                                    ),
                                                ),
                                                SelectColumn(
                                                    Filter(
                                                        Filter(
                                                            Filter(e1, e1_b2), e1_b2
                                                        ),
                                                        SelectColumn(
                                                            Filter(e1, e1_b2), "b1"
                                                        ),
                                                    ),
                                                    "b2",
                                                ),
                                            ),
                                            SelectColumn(
                                                Filter(
                                                    Sort(
                                                        Filter(
                                                            Filter(
                                                                Filter(
                                                                    Filter(e1, e1_b2),
                                                                    e1_b2,
                                                                ),
                                                                SelectColumn(
                                                                    Filter(e1, e1_b2),
                                                                    "b1",
                                                                ),
                                                            ),
                                                            SelectColumn(
                                                                Filter(e1, e1_b2), "b2"
                                                            ),
                                                        ),
                                                        SelectColumn(
                                                            Filter(
                                                                Filter(
                                                                    Filter(e1, e1_b2),
                                                                    e1_b2,
                                                                ),
                                                                SelectColumn(
                                                                    Filter(e1, e1_b2),
                                                                    "b1",
                                                                ),
                                                            ),
                                                            "s1",
                                                        ),
                                                    ),
                                                    SelectColumn(
                                                        Filter(
                                                            Filter(
                                                                Filter(e1, e1_b2), e1_b2
                                                            ),
                                                            SelectColumn(
                                                                Filter(e1, e1_b2), "b1"
                                                            ),
                                                        ),
                                                        "b2",
                                                    ),
                                                ),
                                                "b1",
                                            ),
                                        ),
                                        Position.LAST,
                                    ),
                                    "d1",
                                ),
                                inline_0_d1,
                            ),
                        ),
                        Function.ToFirstOfYear(inline_0_d1),
                    )
                )
            )
        ),
        Case(
            {
                Function.In(
                    Function.IsNull(inline_0_i2),
                    AggregateByPatient.CombineAsSet(Function.GE(e1_s1, e1_s2)),
                ): Function.DateAddYears(
                    Value(datetime.date(2014, 5, 12)), Function.CastToInt(inline_0_i2)
                ),
                Function.IsNull(Function.Subtract(inline_0_i2, inline_1_i1)): Value(
                    datetime.date(2018, 10, 24)
                ),
            },
            default=None,
        ),
    )
)

data = [
    {
        "type": data_setup.P0,
        "patient_id": 9,
        "i1": 3,
        "i2": 9,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2020, 11, 24),
        "d2": datetime.date(2017, 4, 7),
        "f1": 0.5,
        "f2": 0.0,
        "s1": "",
        "s2": "cb",
    },
    {
        "type": data_setup.P0,
        "patient_id": 5,
        "i1": 9,
        "i2": 9,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2013, 7, 5),
        "d2": datetime.date(2019, 9, 25),
        "f1": 5.960464477539063e-08,
        "f2": 5.960464477539063e-08,
        "s1": "bc",
        "s2": "b",
    },
    {
        "type": data_setup.P0,
        "patient_id": 4,
        "i1": 8,
        "i2": 8,
        "b1": True,
        "b2": False,
        "d1": datetime.date(2015, 11, 11),
        "d2": datetime.date(2014, 6, 13),
        "f1": 0.0,
        "f2": 0.0,
        "s1": "a",
        "s2": "a",
    },
    {
        "type": data_setup.P0,
        "patient_id": 8,
        "i1": 3,
        "i2": 6,
        "b1": False,
        "b2": True,
        "d1": datetime.date(2017, 2, 11),
        "d2": datetime.date(2020, 4, 27),
        "f1": 1.1920928955078125e-07,
        "f2": 0.5,
        "s1": "",
        "s2": "a",
    },
    {
        "type": data_setup.P0,
        "patient_id": 7,
        "i1": 0,
        "i2": 10,
        "b1": True,
        "b2": False,
        "d1": datetime.date(2010, 3, 3),
        "d2": datetime.date(2016, 7, 23),
        "f1": 5.85546875,
        "f2": 0.0,
        "s1": "a",
        "s2": "cb",
    },
    {
        "type": data_setup.P0,
        "patient_id": 6,
        "i1": 7,
        "i2": 10,
        "b1": False,
        "b2": True,
        "d1": datetime.date(2018, 11, 15),
        "d2": datetime.date(2013, 1, 15),
        "f1": 1.0,
        "f2": 0.0,
        "s1": "c",
        "s2": "aab",
    },
    {
        "type": data_setup.P0,
        "patient_id": 10,
        "i1": 3,
        "i2": 0,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2010, 7, 6),
        "d2": datetime.date(2015, 12, 29),
        "f1": 1.0,
        "f2": 0.0,
        "s1": "",
        "s2": "c",
    },
    {
        "type": data_setup.P0,
        "patient_id": 2,
        "i1": 1,
        "i2": 0,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2010, 2, 17),
        "d2": datetime.date(2010, 11, 30),
        "f1": 1.5,
        "f2": 1.416015625,
        "s1": "cba",
        "s2": "c",
    },
    {
        "type": data_setup.P0,
        "patient_id": 3,
        "i1": 9,
        "i2": 9,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2020, 11, 4),
        "d2": datetime.date(2018, 5, 4),
        "f1": 9.375,
        "f2": 0.0,
        "s1": "c",
        "s2": "",
    },
    {
        "type": data_setup.P0,
        "patient_id": 1,
        "i1": 6,
        "i2": 10,
        "b1": True,
        "b2": False,
        "d1": datetime.date(2020, 1, 20),
        "d2": datetime.date(2018, 12, 26),
        "f1": 0.0,
        "f2": 0.0,
        "s1": "cb",
        "s2": "b",
    },
    {
        "type": data_setup.P1,
        "patient_id": 10,
        "i1": 8,
        "i2": 8,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2012, 4, 21),
        "d2": datetime.date(2014, 12, 29),
        "f1": 0.0,
        "f2": 10.0,
        "s1": "c",
        "s2": "acc",
    },
    {
        "type": data_setup.P1,
        "patient_id": 1,
        "i1": 5,
        "i2": 9,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2014, 6, 7),
        "d2": datetime.date(2012, 8, 18),
        "f1": 0.0,
        "f2": 0.0,
        "s1": "ab",
        "s2": "bba",
    },
    {
        "type": data_setup.P1,
        "patient_id": 7,
        "i1": 7,
        "i2": 9,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2017, 7, 1),
        "d2": datetime.date(2015, 12, 18),
        "f1": 8.375,
        "f2": 6.103515625e-05,
        "s1": "",
        "s2": "",
    },
    {
        "type": data_setup.P1,
        "patient_id": 5,
        "i1": 9,
        "i2": 10,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2016, 7, 19),
        "d2": datetime.date(2015, 12, 16),
        "f1": 0.0,
        "f2": 0.0,
        "s1": "ab",
        "s2": "",
    },
    {
        "type": data_setup.P1,
        "patient_id": 2,
        "i1": 6,
        "i2": 3,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2016, 7, 28),
        "d2": datetime.date(2015, 7, 25),
        "f1": 0.0,
        "f2": 0.0,
        "s1": "c",
        "s2": "",
    },
    {
        "type": data_setup.P1,
        "patient_id": 3,
        "i1": 7,
        "i2": 9,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2020, 3, 1),
        "d2": datetime.date(2016, 3, 30),
        "f1": 0.0,
        "f2": 2.0,
        "s1": "aac",
        "s2": "",
    },
    {
        "type": data_setup.P1,
        "patient_id": 8,
        "i1": 2,
        "i2": 7,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2013, 11, 9),
        "d2": datetime.date(2019, 3, 16),
        "f1": 0.0,
        "f2": 5.640625,
        "s1": "cba",
        "s2": "ba",
    },
    {
        "type": data_setup.E0,
        "patient_id": 2,
        "i1": 5,
        "i2": 1,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2014, 9, 2),
        "d2": datetime.date(2015, 7, 20),
        "f1": 1.0013580322265625e-05,
        "f2": 6.103515625e-05,
        "s1": "bcb",
        "s2": "a",
    },
    {
        "type": data_setup.E0,
        "patient_id": 8,
        "i1": 4,
        "i2": 5,
        "b1": False,
        "b2": True,
        "d1": datetime.date(2015, 7, 23),
        "d2": datetime.date(2013, 2, 15),
        "f1": 0.0,
        "f2": 0.54541015625,
        "s1": "acb",
        "s2": "b",
    },
    {
        "type": data_setup.E0,
        "patient_id": 2,
        "i1": 0,
        "i2": 1,
        "b1": False,
        "b2": True,
        "d1": datetime.date(2012, 12, 30),
        "d2": datetime.date(2014, 11, 29),
        "f1": 5.960464477539063e-08,
        "f2": 4.40234375,
        "s1": "cb",
        "s2": "abb",
    },
    {
        "type": data_setup.E0,
        "patient_id": 8,
        "i1": 3,
        "i2": 3,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2019, 5, 9),
        "d2": datetime.date(2020, 5, 25),
        "f1": 0.0,
        "f2": 0.0,
        "s1": "",
        "s2": "",
    },
    {
        "type": data_setup.E0,
        "patient_id": 2,
        "i1": 0,
        "i2": 9,
        "b1": True,
        "b2": False,
        "d1": datetime.date(2018, 8, 23),
        "d2": datetime.date(2016, 9, 19),
        "f1": 8.984375,
        "f2": 9.953125,
        "s1": "a",
        "s2": "bca",
    },
    {
        "type": data_setup.E0,
        "patient_id": 9,
        "i1": 9,
        "i2": 9,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2013, 9, 21),
        "d2": datetime.date(2013, 11, 26),
        "f1": 0.0,
        "f2": 11.0,
        "s1": "c",
        "s2": "",
    },
    {
        "type": data_setup.E0,
        "patient_id": 4,
        "i1": 4,
        "i2": 10,
        "b1": True,
        "b2": False,
        "d1": datetime.date(2018, 5, 6),
        "d2": datetime.date(2017, 7, 6),
        "f1": 10.0,
        "f2": 1.099609375,
        "s1": "a",
        "s2": "aac",
    },
    {
        "type": data_setup.E0,
        "patient_id": 5,
        "i1": 2,
        "i2": 0,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2016, 9, 4),
        "d2": datetime.date(2017, 7, 14),
        "f1": 0.333251953125,
        "f2": 0.0,
        "s1": "cc",
        "s2": "",
    },
    {
        "type": data_setup.E0,
        "patient_id": 3,
        "i1": 8,
        "i2": 3,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2010, 9, 10),
        "d2": datetime.date(2016, 9, 26),
        "f1": 0.333251953125,
        "f2": 0.0,
        "s1": "c",
        "s2": "",
    },
    {
        "type": data_setup.E0,
        "patient_id": 7,
        "i1": 3,
        "i2": 9,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2011, 5, 18),
        "d2": datetime.date(2019, 8, 31),
        "f1": 2.0,
        "f2": 7.23828125,
        "s1": "b",
        "s2": "c",
    },
    {
        "type": data_setup.E1,
        "patient_id": 1,
        "i1": 7,
        "i2": 10,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2018, 5, 11),
        "d2": datetime.date(2013, 8, 27),
        "f1": 6.103515625e-05,
        "f2": 0.0,
        "s1": "",
        "s2": "",
    },
    {
        "type": data_setup.E1,
        "patient_id": 8,
        "i1": 2,
        "i2": 5,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2010, 5, 18),
        "d2": datetime.date(2019, 6, 6),
        "f1": 0.5,
        "f2": 0.0,
        "s1": "a",
        "s2": "b",
    },
    {
        "type": data_setup.E1,
        "patient_id": 4,
        "i1": 8,
        "i2": 6,
        "b1": True,
        "b2": False,
        "d1": datetime.date(2015, 11, 8),
        "d2": datetime.date(2015, 12, 19),
        "f1": 1.2724609375,
        "f2": 1.099609375,
        "s1": "a",
        "s2": "",
    },
    {
        "type": data_setup.E1,
        "patient_id": 1,
        "i1": 6,
        "i2": 9,
        "b1": False,
        "b2": False,
        "d1": datetime.date(2020, 2, 23),
        "d2": datetime.date(2017, 12, 25),
        "f1": 0.0,
        "f2": 4.5859375,
        "s1": "bcc",
        "s2": "bbb",
    },
    {
        "type": data_setup.E1,
        "patient_id": 8,
        "i1": 8,
        "i2": 2,
        "b1": False,
        "b2": True,
        "d1": datetime.date(2011, 3, 26),
        "d2": datetime.date(2017, 5, 12),
        "f1": 1.0,
        "f2": 10.0,
        "s1": "b",
        "s2": "ab",
    },
    {
        "type": data_setup.E1,
        "patient_id": 6,
        "i1": 9,
        "i2": 3,
        "b1": True,
        "b2": True,
        "d1": datetime.date(2020, 8, 10),
        "d2": datetime.date(2011, 1, 12),
        "f1": 1.0,
        "f2": 0.0,
        "s1": "",
        "s2": "a",
    },
]
```
</details>

Failure looks like:
![image](https://github.com/opensafely-core/ehrql/assets/19630/adeab177-5afb-4779-83d7-499c67fe1d05)

Minimal (or at least a lot smaller) reproducing example:
```py
from ehrql.query_model.nodes import (
    AggregateByPatient,
    Case,
    Column,
    Function,
    SelectColumn,
    SelectTable,
    TableSchema,
    Value,
)
from tests.generative.test_query_model import data_setup


e0 = SelectTable(
    "e0",
    TableSchema(),
)

e1 = SelectTable(
    "e1",
    TableSchema(
        s1=Column(str),
    ),
)

e1_s1 = SelectColumn(e1, "s1")

a_in_e1_s1 = Function.In(Value("a"), AggregateByPatient.CombineAsSet(e1_s1))

population = Function.Or(
    AggregateByPatient.Exists(e0),
    AggregateByPatient.Exists(e1),
)

variable = Function.MinimumOf(
    (
        Case(
            {
                a_in_e1_s1: Value(0),
            },
            default=None,
        ),
    )
)

data = [
    {"type": data_setup.E0, "patient_id": 1},
    {"type": data_setup.E1, "patient_id": 2, "s1": "a"},
]
```

Failure looks like:
```
AssertionError: Mismatch between in_memory and mssql
assert equals failed
  [                     [                    
    {                     {                  
      'patient_id': 1,      'patient_id': 1, 
      'v': None,            'v': 0,          
    },                    },                 
    {                     {                  
      'patient_id': 2,      'patient_id': 2, 
      'v': 0,               'v': 0,          
    },                    },                 
  ]                     ]
```

Note this is specifically a failure of the MSSQL engine: the other SQL engines pass. I imagine it's something to do with the interaction between `MinimumOf` (which has the weird ScalarSelectAggregation implementation in MSSQL) and `CombineAsSet`.

--- a/ehrql/query_engines/base_sql.py
+++ b/ehrql/query_engines/base_sql.py
@@ -267,8 +267,14 @@ class BaseSQLQueryEngine(BaseQueryEngine):
         # requires referencing the patient_id column of the outer query, but we don't
         # yet know what this is so we use a placeholder column reference which can get
         # replaced later (see `replace_placeholder_references()`).
-        rhs = sqlalchemy.select(table.c.value).where(
-            table.c.patient_id == PLACEHOLDER_PATIENT_ID
+        rhs = (
+            sqlalchemy.select(table.c.value).where(
+                table.c.patient_id == PLACEHOLDER_PATIENT_ID
+            )
+            # Tell SQLAlchemy that the patient ID table should be correlated (because we
+            # don't yet have a reference to this table we have to do this backwards by
+            # telling to correlate everything _except_ the other table).
+            .correlate_except(table)
         )
         lhs = self.get_expr(node.lhs)
         return lhs.in_(rhs)

--- a/tests/integration/test_query_engines.py
+++ b/tests/integration/test_query_engines.py
@@ -295,20 +295,29 @@ def test_population_is_correctly_evaluated_for_containment_queries(engine):
 
 def test_horizontal_aggregation_wrapping_a_series_containment_query_works(engine):
     # Horizontal aggregations in the MSSQL engine are a little odd, and the gentests
-    # exposed a bug here which we don't want to reoccur
+    # exposed a couple of bugs here which we don't want to reoccur
     dataset = create_dataset()
-    dataset.define_population(patients.exists_for_patient())
-    dataset.value = minimum_of(
-        when(patients.i.is_in(events.i)).then(0).otherwise(1),
-        2,
+    dataset.define_population(events.exists_for_patient())
+    dataset.match = minimum_of(
+        when(patients.i.is_in(events.i)).then("T").otherwise("F"), "X"
     )
 
     engine.populate(
         {
-            patients: [{"patient_id": 1}],
-            events: [{"patient_id": 1}],
+            patients: [
+                # This patient should match because i=3 occurs in their events
+                {"patient_id": 1, "i": 3},
+                # This patient should not match
+                {"patient_id": 2, "i": 3},
+            ],
+            events: [
+                {"patient_id": 1, "i": 3},
+                {"patient_id": 2, "i": 0},
+            ],
         }
     )
+
     assert engine.extract(dataset) == [
-        {"patient_id": 1, "value": 1},
+        {"patient_id": 1, "match": "T"},
+        {"patient_id": 2, "match": "F"},
     ]

